src/canopy/knowledge_base/knowledge_base.py

```diff
@@ -437,20 +439,31 @@ def query(self,
         results = [self._query_index(q,
                                      global_metadata_filter,
                                      namespace) for q in queries]
-        results = self._reranker.rerank(results)
+        ranked_results = self._reranker.rerank(results)
 
         return [
             QueryResult(
-                query=r.query,
+                query=rr.query,
                 documents=[
                     DocumentWithScore(
                         **d.dict(exclude={
-                            'values', 'sparse_values', 'document_id'
+                            'document_id'
                         })
                     )
-                    for d in r.documents
-                ]
-            ) for r in results
+                    for d in rr.documents
+                ],
+                debug_info={"db_result": QueryResult(
+                    query=r.query,
+                    documents=[
+                        DocumentWithScore(
+                            **d.dict(exclude={
+                                'document_id'
+                            })
+                        )
+                        for d in r.documents
+                    ]
+                )} if CE_DEBUG_INFO else {}
+            ) for rr, r in zip(ranked_results, results)
```

Comment:
The zip adds a strong assumption about the legnth of ranked results and results. It's true for now that both are in the same length but I suggest to either assert for this or use `zip_longest` or something like that (not sure with python version supports this funciton)

Reply:
Added an assert